packages/core/tracing/src/components/trace-viewer/SpanAttribute.vue

```diff
@@ -0,0 +1,153 @@
+
+const props = defineProps({
+  span: {
+    type: Object as PropType<Span>,
+    required: true,
+  },
+  keyValue: {
+    type: Object as PropType<IKeyValue>,
+    required: true,
+  },
+})
+
+const config = inject<TraceViewerConfig>(TRACE_VIEWER_CONFIG)!
+
+// TODO: This formatter is simply implemented for now
+const formattedValue = computed(() => {
+  const value = unwrapAnyValue(props.keyValue.value)
+
+  if (typeof value === 'string') {
+    return value
+  }
+
+  if (typeof value === 'boolean' || typeof value === 'number') {
+    return value.toString()
+  }
+
+  if (value instanceof Uint8Array) {
+    return JSON.stringify(Array.from(value))
+  }
+
+  return JSON.stringify(value)
+})
+
+// These are keys of attributes whose values are IDs of entities like services, routes, consumers, etc.
+const ENTITY_LINKED_ATTRIBUTE_KEYS = [
+  SPAN_ATTRIBUTES.KONG_SERVICE_ID.name,
+  SPAN_ATTRIBUTES.KONG_ROUTE_ID.name,
+  SPAN_ATTRIBUTES.KONG_CONSUMER_ID.name,
+  SPAN_ATTRIBUTES.KONG_PLUGIN_ID.name,
+  SPAN_ATTRIBUTES.KONG_UPSTREAM_ID.name,
+]
+
+const entityLink = ref<string | undefined>(undefined)
+const entityLinkData = shallowRef<EntityLinkData | undefined>(undefined)
+
+watchEffect(() => {
+  let entity: { entity: string; entityId: string } | undefined
+
+  const value = props.keyValue.value.stringValue
+  if (typeof value === 'string' && value && value !== SPAN_ATTRIBUTE_VALUE_UNKNOWN) {
+    switch (props.keyValue.key) {
+      case SPAN_ATTRIBUTES.KONG_SERVICE_ID.name: {
+        entity = {
+          entity: 'services',
+          entityId: value,
+        }
+        break
+      }
+      case SPAN_ATTRIBUTES.KONG_ROUTE_ID.name: {
+        entity = {
+          entity: 'routes',
+          entityId: value,
+        }
+        break
+      }
+      case SPAN_ATTRIBUTES.KONG_CONSUMER_ID.name: {
+        entity = {
+          entity: 'consumers',
+          entityId: value,
+        }
+        break
+      }
+      case SPAN_ATTRIBUTES.KONG_PLUGIN_ID.name: {
+        entity = {
+          entity: 'plugins',
+          entityId: value,
+        }
+        break
+      }
+      case SPAN_ATTRIBUTES.KONG_UPSTREAM_ID.name: {
+        entity = {
+          entity: 'upstreams',
+          entityId: value,
+        }
+        break
+      }
+    }
+  }
+
+  if (!entity) {
+    entityLink.value = undefined
+    entityLinkData.value = undefined
+    return
+  }
+
+  let plugin: string | undefined
+  if (props.keyValue.key === SPAN_ATTRIBUTES.KONG_PLUGIN_ID.name) {
+    // We will need to parse the plugin name from the span name
+    plugin = getPhaseAndPlugin(props.span.name)?.[1]
+  }
+
+  entityLink.value = config.buildEntityLink?.(entity.entity, entity.entityId, plugin)
+  entityLinkData.value = {
+    id: entity.entityId,
+    label: entity.entityId, // Will be populated asynchronously
+  }
+
+  const controller = new AbortController()
+  config.getEntityLinkData?.(entity.entity, entity.entityId, controller.signal)
```

Comment:
Do we need some error handling here?

Reply:
As we may need the host app to notify the user about the errors, the host app will have to pass in an error handler/listener or we will have to emit error events to make the host app aware of them. 

I talked to Yi this morning and we both agree that we can let `getEntityLinkData` handle the errors and show visual UI elements just in the host app, without surfacing them in components here and throwing them back to the host app. This would also prevent us from being aware of too many details in the host app.

I have updated the comments for TraceViewerConfig and added a try-catch here in case the host app forgot to do so.

What do you think?

Reply:
I think it's good enough 👍